roles/matrix-synapse/templates/synapse/homeserver.yaml.j2

```diff
@@ -1039,6 +1039,27 @@ media_store_path: "/matrix-media-store-parent/{{ matrix_synapse_media_store_dire
 #    config:
 #       directory: /mnt/some/other/directory
 
+# Matrix S3 Storage Provider
+{% if matrix_synapse_s3_media_store_enabled %}
+media_storage_providers:
+- module: s3_storage_provider.S3StorageProviderBackend
+  # In the context of this module the local store means the S3 Bucket
+  # and the remote store the local volume ¯\_(ツ)_/¯ (i swaped it for convenience)
```

Comment:
While swapping does make some sense (`store_local` being the S3 bucket is weird indeed..), I think it may introduce some confusion sometimes.

We normally name configuration keys as they appear upstream (`matrix_SERVICE_SETTING`), so someone looking at `roles/matrix-synapse/defaults/main.yml` and configuring `matrix_synapse_s3_media_store_remote` may think they're influencing the `store_remote` configuration property. In fact, they're influencing `store_local`.

Since both things are confusing, we should probably do something else -- using completely different variables (`matrix_synapse_s3_media_store_on_s3` and `matrix_synapse_s3_media_store_on_local_filesystem`, for example).

Keeping your comment about the meaning of `store_local` and `store_remote` will still be helpful, so I think we should leave it there.

The comments around `matrix_synapse_s3_media_store_on_s3` and `matrix_synapse_s3_media_store_on_local_filesystem` in `roles/matrix-synapse/defaults/main.yml` may point out which configuration key they influence.

Reply:
Renamed the media options.